src/sentry/static/sentry/app/components/events/eventRelatedEvents.tsx

```diff
@@ -0,0 +1,152 @@
+import {ALL_ACCESS_PROJECTS} from 'app/constants/globalSelectionHeader';
+import {getTraceDateTimeRange} from 'app/components/events/interfaces/spans/utils';
+
+import RelatedEvents from './relatedEvents';
+import EmptyState from './relatedEvents/emptyState';
+import {getCurrentLocation} from './relatedEvents/utils';
+import getDiscoverButton from './relatedEvents/getDiscoverButton';
+
+type Props = {
+  location: Location;
+  event: Event;
+  organization: Organization;
+};
+
+type State = {
+  isLoading: boolean;
+  orgFeatures: Set<string>;
+  orgSlug: string;
+  eventView?: EventView;
+};
+
+class EventRelatedEvents extends React.Component<Props, State> {
+  state: State = {
+    isLoading: true,
+    orgFeatures: new Set(this.props.organization.slug),
+    orgSlug: this.props.organization.slug,
+  };
+
+  componentDidMount() {
+    this.getEventView();
+  }
+
+  getEventView() {
+    const {event, organization} = this.props;
+
+    const traceID = event.contexts?.trace?.trace_id;
+
+    if (!traceID) {
+      this.setState({isLoading: false});
+      return;
+    }
+
+    const orgFeatures = new Set(organization.features);
+    const dateCreated = moment(event.dateCreated).valueOf() / 1000;
+    const pointInTime = event.dateReceived
+      ? moment(event.dateReceived).valueOf() / 1000
+      : dateCreated;
+
+    const {start, end} = getTraceDateTimeRange({
+      start: pointInTime,
+      end: pointInTime,
+    });
+
+    const eventFromSavedQuery = EventView.fromSavedQuery({
+      id: undefined,
+      name: `Events with Trace ID ${traceID}`,
+      fields: [
+        'title',
+        'event.type',
+        'project',
+        'project.id',
+        'trace.span',
+        'timestamp',
+        'lastSeen',
+        'issue',
+      ],
+      orderby: '-timestamp',
+      query: `trace:${traceID}`,
+      projects: orgFeatures.has('global-views')
+        ? [ALL_ACCESS_PROJECTS]
+        : [Number(event.projectID)],
+      version: 2,
+      start,
+      end,
+    });
+
+    this.setState({eventView: eventFromSavedQuery, isLoading: false});
+  }
+
+  renderEmptyState(message?: string) {
+    return <EmptyState message={message} />;
+  }
+
+  render() {
+    const {location, event} = this.props;
+    const {isLoading, eventView, orgSlug, orgFeatures} = this.state;
+
+    if (isLoading) {
+      return <LoadingIndicator />;
+    }
+
+    const currentLocation = getCurrentLocation();
+
+    return (
+      <EventDataSection
+        type="related-events"
+        title={t('Related Events')}
+        actions={getDiscoverButton({orgSlug, currentLocation, orgFeatures, eventView})}
+      >
+        {eventView ? (
+          <DiscoverQuery location={location} eventView={eventView} orgSlug={orgSlug}>
+            {discoverData => {
+              if (discoverData.isLoading) {
+                return <LoadingIndicator />;
+              }
+
+              if (!discoverData.tableData?.data) {
```

Comment:
I think it might be better to check `orgFeatures` for `discover-basic` and return `null` if it's not available.

Reply:
Maybe `performance-view` instead?

Reply:
After talking to Filip, we decided to make this feature available to all customers, with the exception of the "Open in discovery button" which should only be available if the user has "discover-basic" 😉